src/storage-client/src/controller.rs

```diff
@@ -523,6 +518,8 @@ pub trait StorageController: Debug {
         updates: Vec<(Row, Diff)>,
     );
 
+    async fn append_status_updates(&mut self, updates: Vec<Row>, type_: IntrospectionType);
```

Comment:
Actually it looks like the only caller of `record_introspection_updates` is now the statement logging -- I wonder if that could be refactored to use `append_status_updates` too? It could be one that doesn't implement `IntrospectionStatusUpdate` which should skip all the checks you implemented against previous status values